app/resources/api/v1/protected/session_resource.rb

```diff
@@ -0,0 +1,28 @@
+class Api::V1::Protected::SessionResource < JSONAPI::Resource
```

Comment:
The `JSONAPI::Resource` is what you call a DTO.  You create a resource for all endpoints.  For example, if we were going to grant public access to this resource we'd have to create it and a controller under the `api/v1/public/...` path.

Reply:
Makes sense.

Reply:
The one problem I'm starting see is at the EmberJs/ReactApp level.  The data adapter down at that level will need to speak to two different endpoints.  That might get complicated.